src/Domain/LeanCode.DomainModels.EF/EFRepository.cs

```diff
@@ -41,12 +41,14 @@ public virtual void Delete(TEntity entity)
 
     public virtual void DeleteRange(IEnumerable<TEntity> entities)
     {
-        foreach (var oc in entities.OfType<IOptimisticConcurrency>())
+        var entitiesList = entities.ToList();
```

Comment:
Let's revert this. `ToList` here forcibly allocates a list, whereas double-enumeration will not, and most of the time it will be cheaper, because someone either passes list already or passes something that is a cheap mapping of list.